File: pdu/python/drone_srv_msgs/pdu_cdr_conv_DroneGetStateRequest.py
from .pdu_pytype_DroneGetStateRequest import DroneGetStateRequest
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_DroneGetStateRequest(writer: CdrWriter, src: DroneGetStateRequest):
    writer.write_string(src.drone_name)


def cdr_body_to_py_DroneGetStateRequest(reader: CdrReader, dst: DroneGetStateRequest):
    dst.drone_name = reader.read_string()
    return dst


def py_to_cdr_DroneGetStateRequest(src: DroneGetStateRequest) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_DroneGetStateRequest(writer, src)
    return writer.bytes()


def cdr_to_py_DroneGetStateRequest(cdr_payload) -> DroneGetStateRequest:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = DroneGetStateRequest()
    return cdr_body_to_py_DroneGetStateRequest(reader, dst)

File: pdu/javascript/drone_srv_msgs/pdu_cdr_conv_DroneGetStateRequest.js
import { DroneGetStateRequest } from './pdu_jstype_DroneGetStateRequest.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduDroneGetStateRequestConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { DroneGetStateRequest } src
     */
    static to_cdr_body(writer, src) {
        writer.write_string(src.drone_name);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { DroneGetStateRequest } dst
     * @returns { DroneGetStateRequest }
     */
    static cdr_body_to_js(reader, dst) {
        dst.drone_name = reader.read_string();
        return dst;
    }

    /**
     * @param { DroneGetStateRequest } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { DroneGetStateRequest }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new DroneGetStateRequest());
    }
}
